File: src/ISAI.Domain/Entities/Memory.cs
using System;

namespace ISAI.Domain.Entities;

public class Memory
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public Guid UserId { get; set; }
    public User? User { get; set; }
    public string Content { get; set; } = string.Empty;
    public string Category { get; set; } = "General"; // Preference, Fact, Work, Personal
    public string Source { get; set; } = "UserExplicit"; // UserExplicit, SystemSuggested
    public string SensitivityLevel { get; set; } = "Normal"; // Normal, Private, Confidential
    public bool IsEnabled { get; set; } = true;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? DeletedAt { get; set; }
}

File: backend/app/schemas/memory.py
from datetime import datetime
from pydantic import BaseModel, ConfigDict

class MemoryCreateRequest(BaseModel):
    content: str
    category: str = "General"
    source: str = "UserExplicit"
    sensitivity_level: str = "Normal"
    is_enabled: bool = True

class MemoryUpdateRequest(BaseModel):
    content: str | None = None
    category: str | None = None
    sensitivity_level: str | None = None
    is_enabled: bool | None = None

class MemoryDto(BaseModel):
    id: str
    user_id: str
    content: str
    category: str
    source: str
    sensitivity_level: str
    is_enabled: bool
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)
